### ISSUE TITLE
Running Windows installer with LauncherOnly=1 should not register the version as installed

### ISSUE THREAD

**Issue by ddf74a51-5f28-44b3-89e3-5080a0864212** [python/cpython#75875]
BPO | [31694](https://bugs.python.org/issue31694)
--- | :---
Nosy | @pfmoore, @tjguk, @zware, @zooba, @uranusjr

<sup>*Note: these values reflect the state of the issue at the time it was migrated and might not reflect the current state.*</sup>

<details><summary>Show more details</summary><p>

GitHub fields:
```python
assignee = 'https://github.com/zooba'
closed_at = None
created_at = <Date 2017-10-04.18:52:39.634>
labels = ['3.7', 'type-bug', 'expert-installation', 'OS-windows']
title = 'Running Windows installer with LauncherOnly=1 should not register the version as installed'
updated_at = <Date 2017-10-05.15:07:43.267>
user = 'https://github.com/uranusjr'
```

bugs.python.org fields:
```python
activity = <Date 2017-10-05.15:07:43.267>
actor = 'steve.dower'
assignee = 'steve.dower'
closed = False
closed_date = None
closer = None
components = ['Installation', 'Windows']
creation = <Date 2017-10-04.18:52:39.634>
creator = 'uranusjr'
dependencies = []
files = []
hgrepos = []
issue_num = 31694
keywords = []
message_count = 3.0
messages = ['303727', '303742', '303774']
nosy_count = 5.0
nosy_names = ['paul.moore', 'tim.golden', 'zach.ware', 'steve.dower', 'uranusjr']
pr_nums = []
priority = 'normal'
resolution = None
stage = None
status = 'open'
superseder = None
type = 'behavior'
url = 'https://bugs.python.org/issue31694'
versions = ['Python 3.7']
```

</p></details>

**Comment by ddf74a51-5f28-44b3-89e3-5080a0864212** [comment]
After running

    python-3.6.2-amd64.exe /quiet LauncherOnly=1

Python 3.6.2 is registered as installed, even though nothing is actually installed.

Running the installer again at this point shows an error indicating “this product is already installed”, and does not show me the usual repair/modify/uninstall options.

I have to run uninstallation with Control Panel before the installer works again.

**Comment by ddf74a51-5f28-44b3-89e3-5080a0864212** [comment]
The “this product is already installed” error is a mistake on my part. I was actually using 3.6.3 (released literally during I was testing this!) to install the launcher, and using 3.6.2 afterwards, hence the error message (because I was trying to modify using an older version).

The main issue however stays. The installer still shouldn’t register the main Python distribution as installed when only the launcher was selected.

**Comment by zooba** [comment]
Agreed, and the underlying cause is also linked to the registration being wrong in some other ways.

The fix requires rearchitecting the installer again, though less dramatically than last time. Won't happen for3.6.4, but we might get 3.7.

**Comment by zooba** [comment]
The installer referred to in this issue has now been removed (#153511), and so this issue is expired. Please use the [Python install manager](https://www.python.org/downloads/latest/pymanager/) to install Python, and file any issues [on the pymanager repo](https://github.com/python/pymanager).
